server/src/main/java/com/vaadin/server/VaadinService.java

```diff
@@ -611,7 +611,7 @@ protected Lock getSessionLock(WrappedSession wrappedSession) {
      * @throws IllegalStateException
      *             if the session is invalidated before it can be locked
      */
-    protected void lockSession(WrappedSession wrappedSession) {
+    protected Lock lockSession(WrappedSession wrappedSession) {
```

Comment:
Could maybe add a code comment or Javadoc mention of the previous version of this method so that someone who gets errors on upgrading can find an explanation by looking at code also and not only by reading release notes.

Reply:
Yes, it was in my plans to add also some note that override of this method is not recommended

Reply:
There's nothing that explains the migration scenario, but only a new unrelated comment that blames the user for doing something that might not be right, based on the assumption that it's only about lock storage but not about how the actual locking is performed.